src/debug_commands.c

```diff
@@ -1402,6 +1430,123 @@ DEBUG_COMMAND(WorkerThreadsSwitch) {
+
+  // Check if we need to enable debug mode
+  validateDebugMode(&debugCtx);
+
+  return RedisModule_ReplyWithSimpleString(ctx, "OK");
+}
+
+DEBUG_COMMAND(setPauseOnScannedDocs) {
+  if (!debugCommandsEnabled(ctx)) {
+    return RedisModule_ReplyWithError(ctx, NODEBUG_ERR);
+  }
+  if (argc != 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+  long long pause_scanned_docs;
+  if (RedisModule_StringToLongLong(argv[2], &pause_scanned_docs) != REDISMODULE_OK) {
+    return RedisModule_ReplyWithError(ctx, "Invalid argument for 'SET_MAX_SCANNED_DOCS'");
+  }
+
+  // Negative maxDocsTBscanned represents no limit
+
+  debugCtx.maxDocsTBscannedPause = (int) pause_scanned_docs;
+
+  // Check if we need to enable debug mode
+  validateDebugMode(&debugCtx);
+
+  return RedisModule_ReplyWithSimpleString(ctx, "OK");
+}
+
+DEBUG_COMMAND(setResume) {
+  if (!debugCommandsEnabled(ctx)) {
+    return RedisModule_ReplyWithError(ctx, NODEBUG_ERR);
+  }
+  if (argc != 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+  long long resume;
+  if (RedisModule_StringToLongLong(argv[2], &resume) != REDISMODULE_OK) {
+    return RedisModule_ReplyWithError(ctx, "Invalid argument for 'SET_RESUME'");
+  }
+
+  debugCtx.pause = !resume;
+
+  return RedisModule_ReplyWithSimpleString(ctx, "OK");
+}
+
+DEBUG_COMMAND(getDebugScannerStatus) {
+  if (!debugCommandsEnabled(ctx)) {
+    return RedisModule_ReplyWithError(ctx, NODEBUG_ERR);
+  }
+  if (argc != 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  IndexLoadOptions lopts = {.nameC = RedisModule_StringPtrLen(argv[2], NULL),
+                            .flags = INDEXSPEC_LOAD_NOTIMERUPDATE};
+
+  StrongRef ref = IndexSpec_LoadUnsafeEx(&lopts);
+  IndexSpec *sp = StrongRef_Get(ref);
+
+  if (!sp) {
+    return RedisModule_ReplyWithError(ctx, "Unknown index name");
+  }
+
+  if (!debugCtx.debugMode) {
+    return RedisModule_ReplyWithError(ctx, "Debug mode is not enabled");
+  }
+
+  if(!(sp->flags & Index_DebugScanner)) {
+    return RedisModule_ReplyWithError(ctx, "Debug mode enabled but scanner is not a debug scanner");
+  }
+
+  // Assuming this file is aware of spec.h, via direct or in-direct include//
+  DebugIndexesScanner *dScanner = (DebugIndexesScanner*)sp->scanner;
+
+  if (!dScanner) {
+    return RedisModule_ReplyWithError(ctx, "Scanner is not initialized");
+  }
+  const char* strs[] = DEBUG_INDEX_SCANNER_STATUS_STRS;
+
+  return RedisModule_ReplyWithSimpleString(ctx, strs[dScanner->status]);
+}
+
+DEBUG_COMMAND(setPauseBeforeScan) {
+  if (!debugCommandsEnabled(ctx)) {
+    return RedisModule_ReplyWithError(ctx, NODEBUG_ERR);
+  }
+  if (argc != 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+  long long pauseBeforeScan;
+  if (RedisModule_StringToLongLong(argv[2], &pauseBeforeScan) != REDISMODULE_OK) {
+    return RedisModule_ReplyWithError(ctx, "Invalid argument for 'SET_PAUSE_BEFORE_SCAN'");
+  }
+
+  debugCtx.pauseBeforeScan = !!pauseBeforeScan;
```

Comment:
If this is a boolean option, consider allowing only `ON`/`OFF` or `TRUE`/`FALSE` (can be case insensitive) instead of numeric values